File: packages/syntax/test/fixtures/sample.py
"""Sample Python file for testing."""

from typing import List, Optional

VERSION = "1.0.0"


class Calculator:
    """A simple calculator class."""

    def __init__(self, value: int = 0):
        """Initialize the calculator."""
        self.value = value

    def add(self, x: int) -> int:
        """Add x to the current value."""
        self.value += x
        return self.value

    def subtract(self, x: int) -> int:
        """Subtract x from the current value."""
        self.value -= x
        return self.value

    @staticmethod
    def multiply(a: int, b: int) -> int:
        """Multiply two numbers."""
        return a * b


def greet(name: str) -> str:
    """Greet someone by name."""
    return f"Hello, {name}!"


async def fetch_data(url: str) -> str:
    """Fetch data from a URL."""
    return "data"

File: packages/syntax/test/fixtures/sample.ts
/**
 * Sample TypeScript file for testing.
 */

import { readFile } from "fs/promises";
import type { Result } from "./types";

export const VERSION = "1.0.0";

/**
 * A simple calculator class.
 */
export class Calculator {
  private value: number;

  constructor(initial: number = 0) {
    this.value = initial;
  }

  /**
   * Add a number to the current value.
   */
  add(x: number): number {
    this.value += x;
    return this.value;
  }

  /**
   * Subtract a number from the current value.
   */
  subtract(x: number): number {
    this.value -= x;
    return this.value;
  }

  getValue(): number {
    return this.value;
  }
}

/**
 * Multiply two numbers.
 */
export function multiply(a: number, b: number): number {
  return a * b;
}

export async function fetchData(url: string): Promise<string> {
  return "data";
}

interface User {
  id: number;
  name: string;
}

type Status = "active" | "inactive";
